client/my-sites/site-settings/form-jetpack-monitor.jsx

```diff
@@ -1,212 +1,199 @@
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+import { isEmpty } from 'lodash';
 
 /**
  * Internal dependencies
  */
-var config = require( 'config' ),
-	Card = require( 'components/card' ),
-	FormCheckbox = require( 'components/forms/form-checkbox' ),
-	FormLabel = require( 'components/forms/form-label' ),
-	formBase = require( './form-base' ),
-	notices = require( 'notices' ),
-	dirtyLinkedState = require( 'lib/mixins/dirty-linked-state' ),
-	SectionHeader = require( 'components/section-header' ),
-	Button = require( 'components/button' );
-import { protectForm } from 'lib/protect-form';
-
-module.exports = protectForm( React.createClass( {
-
-	displayName: 'SiteSettingsFormJetpackMonitor',
-
-	mixins: [ dirtyLinkedState, formBase ],
-
-	getSettingsFromSite: function( site ) {
-		var settings = {};
-
-		site = site || this.props.site;
-		settings.fetchingSettings = site.fetchingSettings;
-
-		return settings;
-	},
-
-	resetState: function() {
-		this.replaceState( {
-			togglingModule: false,
-			emailNotifications: false,
-			wpNoteNotifications: false
-		} );
-	},
+import config from 'config';
+import Card from 'components/card';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
+import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
+import SectionHeader from 'components/section-header';
+import InfoPopover from 'components/info-popover';
+import ExternalLink from 'components/external-link';
+import QueryJetpackModules from 'components/data/query-jetpack-modules';
+import QuerySiteMonitorSettings from 'components/data/query-site-monitor-settings';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { updateSiteMonitorSettings } from 'state/sites/monitor/actions';
+import { recordGoogleEvent } from 'state/analytics/actions';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import {
+	getSiteMonitorSettings,
+	isActivatingJetpackModule,
+	isDeactivatingJetpackModule,
+	isFetchingJetpackModules,
+	isJetpackModuleActive,
+	isMonitorSettingsUpdateSuccessful,
+	isRequestingSiteMonitorSettings,
+	isUpdatingSiteMonitorSettings
+} from 'state/selectors';
+
+class SiteSettingsFormJetpackMonitor extends Component {
+	state = {};
+
+	componentWillReceiveProps( nextProps ) {
+		if ( isEmpty( this.state ) && nextProps.monitorSettings ) {
+			this.setState( nextProps.monitorSettings );
+		}
+	}
 
-	getModuleStatus: function( site ) {
-		site = site || this.props.site;
-
-		site.fetchMonitorSettings( ( function( error, data ) {
-			if ( ! error ) {
-				this.setState( {
-					emailNotifications: data.settings.email_notifications,
-					wpNoteNotifications: data.settings.wp_note_notifications
-				} );
-			} else {
-				debug( 'error getting Monitor settings', error );
-			}
-		} ).bind( this ) );
+	recordEvent = ( event ) => {
+		return () => {
+			this.props.trackEvent( event );
+		};
+	}
```

Comment:
Whatever we choose we should be consistent. The majority of other class property definitions I've seen in the codebase do use `;` like propTypes, defaultProps, and state. I personally like being explicit here, though it doesn't matter much if we pick without, since the code is transpiled.

eg.
```jsx
static propTypes = {
    isVisible: PropTypes.bool,
};

```